### PR TITLE
Fix merge checks

### PR DESCRIPTION
**Pull request by eccles**
Use virtual environement using 'trap deactivate EXIT'

Merging requires that the ci git hub action passes all checks and also passes functional tests a.k.a registration-demo.

The github action of releasing must only be executed when releasing as wheels in pypi are immutable. Otherwise we would have to bump release versions if the releases fail. Once code is merged it is releasable.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,28 +28,28 @@ jobs:
...
             echo "there are $modified files that must be reformatted"
             echo "DISABLED guard due to mismatch with local environment"
             # exit 1
         fi
-        python3 -m unittest
       shell: bash
-    - name: Run type-hint checks
-      if: ${{ matrix.python-version != '3.12' }}
+
+    - name: Unit tests
```
Comment (robinbryce) [inline .github/workflows/ci.yml:44]: very minor: Do we really want to call these unit tests ? It is true they are being run using the unittest framework, but they are actually *end to end* tests for the whole scitt solution on our platform. If there is a consistent thing to do here lets do that. but its small beer

Comment (eccles) [reply .github/workflows/ci.yml:44]: yes - I think we should rename them

Comment (eccles) [reply .github/workflows/ci.yml]: renamed to 'tests'

**Review by robinbryce — APPROVED** [review]
LGTM

**Pull request merged**